modules/core/src/main/java/org/apache/synapse/config/xml/MediatorPropertyFactory.java

```diff
@@ -82,7 +83,11 @@
                     throw new SynapseException(msg);
 
                 } else {
-                    prop.setValue(attValue.getAttributeValue());
+                    String attributeValue = attValue.getAttributeValue();
+                    prop.setValue(attributeValue);
+                    if (mediator != null) {
```

Comment:
Can we avoid this null check? Whether the mediator is null or not is determined at the compile time itself.

Reply:
This mediator parameter is introduced to distinguish the usage of method at the class mediator. This is a common method used for log, xslt, and endpoint factories too. By default, log. xslt, endpoint have mediator parameter as null. Usage of this method will be derived at runtime. So, we need to have this null check.